Workspace/src/xbox_controller/src/teleop_node.cpp

```diff
@@ -24,15 +24,34 @@ class StateSelect {
     joy_sub =
         n.subscribe<sensor_msgs::Joy>("joy", 1, &StateSelect::callback, this);
 
-    // Publisher to three separate branches
+    // Publisher to four separate branches
     drive_pub = n.advertise<sensor_msgs::Joy>("drive_joy", 1);
     science_pub = n.advertise<sensor_msgs::Joy>("science_joy", 1);
     arm_pub = n.advertise<sensor_msgs::Joy>("arm_joy", 1);
+    autonomy_pub = n.advertise<sensor_msgs::Joy>("autonomy_joy", 1);
 
     // Logic Variables Initialization
     i = DRIVE;       // Starts at the drive branch
     pressed = false; // Initially, the change button is assumed to be NOT
                      // PRESSED
+    
+    // populate timestamp, axes and buttons of empty joy msg, make into function?
+    empty_joy_msg.header.stamp = ros::Time().now();
+
+    // get size of buttons and axes arrays
+    int num_buttons = empty_joy_msg.buttons.size();

```

Comment:
Will the empty Joy message have a default size? Would we need to resize it based on how large the Xbox joy message is?

Reply:
Not sure exactly, but regardless of its size the for loops will populate every single button/axis with 0s so I don't think it will matter. Based on what I can find online I think there is only one type of joy message so hopefully the size changing won't be a factor and the default is just the regular size. Hopefully we get a chance to test it again on the actual rover after exams to see.